File: backend/src/main/java/com/university/cms/service/CourseService.java
package com.university.cms.service;

import com.university.cms.entity.Course;
import com.university.cms.entity.User;
import com.university.cms.repository.CourseRepository;
import com.university.cms.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class CourseService {
    
    @Autowired
    private CourseRepository courseRepository;
    
    @Autowired
    private UserRepository userRepository;
    
    public List<Course> getAllCourses() {
        return courseRepository.findAll();
    }
    
    public Optional<Course> getCourseById(Long id) {
        return courseRepository.findById(id);
    }
    
    public Course createCourse(Course course) {
        if (courseRepository.existsByCode(course.getCode())) {
            throw new RuntimeException("Course code already exists");
        }
        return courseRepository.save(course);
    }
    
    public Course updateCourse(Long id, Course courseDetails) {
        Course course = courseRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Course not found"));
        
        course.setCode(courseDetails.getCode());
        course.setTitle(courseDetails.getTitle());
        course.setDescription(courseDetails.getDescription());
        course.setCredits(courseDetails.getCredits());
        course.setTeacher(courseDetails.getTeacher());
        
        return courseRepository.save(course);
    }
    
    public void deleteCourse(Long id) {
        courseRepository.deleteById(id);
    }
    
    public List<Course> getCoursesByTeacher(Long teacherId) {
        return courseRepository.findByTeacherId(teacherId);
    }
    
    public Course assignTeacherToCourse(Long courseId, Long teacherId) {
        Course course = courseRepository.findById(courseId)
                .orElseThrow(() -> new RuntimeException("Course not found"));
        
        User teacher = userRepository.findById(teacherId)
                .orElseThrow(() -> new RuntimeException("Teacher not found"));
        
        course.setTeacher(teacher);
        return courseRepository.save(course);
    }
     public long getCourseCount() {
        return courseRepository.count();
    }
}

File: frontend/src/services/courseService.js
import axios from 'axios';

const API_URL = process.env.REACT_APP_API_URL || 'http://localhost:8080/api';

const courseService = {
  async getAllCourses() {
    const response = await axios.get(`${API_URL}/courses`);
    return response.data;
  },

  async getCourseById(id) {
    const response = await axios.get(`${API_URL}/courses/${id}`);
    return response.data;
  },

  async createCourse(courseData) {
    const response = await axios.post(`${API_URL}/courses`, courseData);
    return response.data;
  },

  async updateCourse(id, courseData) {
    const response = await axios.put(`${API_URL}/courses/${id}`, courseData);
    return response.data;
  },

  async deleteCourse(id) {
    const response = await axios.delete(`${API_URL}/courses/${id}`);
    return response.data;
  },

  async getCoursesByTeacher(teacherId) {
    const response = await axios.get(`${API_URL}/courses/teacher/${teacherId}`);
    return response.data;
  }
};

export default courseService;
